doc/architecture-decisions/0012-routecontroller-route-deletion-finalizer.md

```diff
@@ -0,0 +1,27 @@
+# 11. Routecontroller Route Deletion Finalizer
+
+Date: 2020-05-11
+
+## Status
+
+Accepted
+
+## Context
+Without a finalizer, we could rely on cascading deletion of child objects when Routes are deleted and `RequeueAfter` to rebuild the virtual services and services  from its routes every `ResyncInterval` seconds. `ResyncInterval` is currently set to 30 seconds.
+
+However, this doesn’t meet our SLO to handle changes within 10 seconds. Cascading deletes don’t work within 10 seconds, so meeting the SLO would require `ResyncInterval` to be less than 10 seconds, which seems unreasonable. 
+
+Cascading deletion alone also doesn’t handle the case of a virtual service being owned by >1 Route. This means a cascading delete cannot update the virtual service’s contents to not include the paths related to that deleted route. So we would need to rely on `RequeueAfter` for those updates, which would be slow.
```

Comment:
To be fair we hadn't even considered relying on `RequeueAfter` in this case, it just so happens that it would probably work. 😛 I would either omit this or that relying on `RequeueAfter` set on a completely different `Route` to react to the deletion of a related `Route` is kind of an anti-pattern.